pkg/messageprocessors/normalizedpayload/uplink.go

```diff
@@ -117,12 +139,68 @@ func minimum[T constraints.Ordered](min T) fieldValidator[T] {
 	}
 }
 
+// exclusiveMinimum returns a field validator that checks the exclusive minimum.
+//nolint:unused,deadcode
+func exclusiveMinimum[T constraints.Ordered](min T) fieldValidator[T] {
+	return func(v T, path string) error {
+		if v <= min {
+			return errFieldExclusiveMinimum.WithAttributes(
+				"path", path,
+				"minimum", min,
+			)
+		}
+		return nil
+	}
+}
+
+// maximum returns a field validator that checks the inclusive maximum.
+func maximum[T constraints.Ordered](max T) fieldValidator[T] {
+	return func(v T, path string) error {
+		if v > max {
+			return errFieldMaximum.WithAttributes(
+				"path", path,
+				"maximum", max,
+			)
+		}
+		return nil
+	}
+}
+
+// exclusiveMaximum returns a field validator that checks the exclusive maximum.
+func exclusiveMaximum[T constraints.Ordered](max T) fieldValidator[T] {
+	return func(v T, path string) error {
+		if v >= max {
+			return errFieldExclusiveMaximum.WithAttributes(
+				"path", path,
+				"maximum", max,
+			)
+		}
+		return nil
+	}
+}
+
 var fieldParsers = map[string]fieldParser{
 	"time": parseTime(func(dst *Measurement) **time.Time { return &dst.Time }),
 	"air":  object(func(dst *Measurement) **Air { return &dst.Air }),
 	"air.temperature": parseNumber(func(dst *Measurement) **float64 { return &dst.Air.Temperature },
 		minimum(-273.15),
 	),
+	"air.relativeHumidity": parseNumber(func(dst *Measurement) **float64 { return &dst.Air.RelativeHumidity },
+		minimum(0.0),
+		maximum(1.0),
+	),
+	"air.pressure": parseNumber(func(dst *Measurement) **float64 { return &dst.Air.Pressure },
```

Comment:
Schema PR related: We call this air here and in the schema, but shouldn't we call this actually environment ?

Think of a sensor that measures the state temperature outside of a factory, vs one that measures the temperature/ RH / pressure of a container / chamber - the validation probably differs between the two, and if the container has for some reasons air in it, it becomes a bit weird.

Reply:
Environment is also relative to the sensor, just like outside a temperature sensor can be in the shade or in the sun.

This positioning and hierarchy is to be done upstream. The sensor is an air temperature sensor, which could be placed on a ship or in a container; the same device model. We're not intending to make the firmware aware whether it's indoor or outdoor. In the digital twin of that ship, the sensors should be modeled correctly.